Industrial Sector Energy Use - nrcan.ipynb

```diff
@@ -0,0 +1,318 @@
+    "\n",
+    "#Concatenate table_id number\n",
+    "def write_table_id(index):\n",
+    "    table_id = 'id_ca_e_' + str(index)\n",
+    "    return table_id\n",
+    "\n",
+    "#add additional columns to table\n",
+    "def add_other_data(df, targetName, table_id):\n",
+    "    df.insert(1, 'Target', targetName)\n",
+    "    df.insert(3, 'Units', 'Petajoules')\n",
+    "    df.insert(4, 'Table ID', table_id)\n",
+    "    return df\n",
+    "\n",
+    "#extract only relevant data\n",
+    "#note locations are hardcodes for nrcan files types -- all are equivalent\n",
+    "def extract_relevant_data(df):\n",
+    "    df = df.drop(df.iloc[:, 2:29], inplace = False, axis = 1)\n",
+    "    df = df.iloc[13:23,:]\n",
+    "    df = df.drop('Back to main menu',1)\n",
+    "    return df\n",
+    "\n",
+    "def extract_data_last_two(df):\n",
+    "    df = df.drop(df.iloc[:, 2:29], inplace = False, axis = 1)\n",
+    "    df = df.iloc[13:23,:]\n",
+    "    df = df.drop(df.iloc[:, 3:6], inplace = False, axis = 1)\n",
+    "    df = df.drop('Back to main menu',1)\n",
+    "    return df\n",
+    "\n",
+    "qgrid.enable()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Loop Through Tables 9 - 67 and append to master df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 209,
+   "metadata": {
+    "ExecuteTime": {
+     "end_time": "2020-12-07T03:03:36.304227Z",
+     "start_time": "2020-12-07T03:03:36.291715Z"
+    },
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "filePath = 'data/idca2017e/id_ca_e_'\n",
+    "net_df = pd.DataFrame(columns=['Source','Target','Value','Units','Table ID'])\n",
+    "\n",
+    "#index through all tables in idca2017e folder\n",
+    "for x in range(9,66):\n",
+    "    fileIndex = str(x)\n",
+    "    path = filePath + fileIndex + '.csv'\n",
+    "    df.iloc[0:0]\n",
+    "    df = pd.read_csv(path, encoding='cp1252', header=0)\n",
+    "    tableName = df.iloc[6,0]\n",
+    "    targetName = substring_table(tableName)\n",
+    "    table_id = write_table_id(fileIndex)\n",
+    "    df = extract_relevant_data(df)\n",
+    "    df = rename_columns(df)\n",
+    "    df = add_other_data(df, targetName, table_id)\n",
+    "    net_df = net_df.append(df)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 210,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "application/vnd.jupyter.widget-view+json": {
+       "model_id": "57ffcf8a495b4b65afcdc9e3bb7fcb30",
+       "version_major": 2,
+       "version_minor": 0
+      },
+      "text/plain": [
+       "QgridWidget(grid_options={'fullWidthRows': True, 'syncColumnCellResize': True, 'forceFitColumns': True, 'defau…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "#index through last two tables which randomly have 3 extra columns at the end\n",
+    "for x in range(66, 68):\n",
```

Comment:
Could add these indices to the above loop by concatenating into a single list of indices`for x in list(range(9,66)) +  list(range(66,68))`

Reply:
What I did is hacky but basically the code that runs through 9-65 breaks on 66 / 67 because they're formatted differently and have extra columns that cause an error. Not clear why they're like that, but I just gave them their own function for data extraction instead of making the others more flexible. So, I would advise not making the proposed change. 

Actually - realizing I probably could have added a dropna but they were being quite finicky for whatever reason! Let me see if it works when I do that. Would be cleaner.